Packs/OpenAI/Integrations/OpenAI/OpenAI.py

```diff
@@ -1,109 +1,230 @@
+        """
+            Enter an instruction and watch the OpenAI API respond with a completion that attempts to match the context
+            or pattern you provided, using the 'completions' endpoint.
+
+            :type prompt: ``str``
+            :param prompt: Instruction
+            :type model: ``str``
+            :param model: The model which will generate the completion.
+            :type temperature: ``float``
+            :param temperature: Controls randomness: Lowering results in less random completions.
+            :type max_tokens: ``int``
+            :param max_tokens: The maximum number of tokens to generate.
+            :type top_p: ``float``
+            :param top_p: Controls Diversity via nucleus sampling
+            :type frequency_penalty: ``int``
+            :param frequency_penalty: How much to penalize new tokens based on their existing frequency in the text so far.
+            :type presence_penalty: ``int``
+            :param presence_penalty: How much to penalize new tokens based on whether they appear in the text so far.
+            :type best_of: ``int``
+            :param best_of: Generates best_of completions server-side and returns the "best"
+            :type stop: ``str``
+            :param stop: The returned text won't contain the stop sequence.
+
+            :return: response of the OpenAI Completion API
+            :rtype: ``dict``
         """
 
+        # available API params for OpenAI: https://platform.openai.com/docs/api-reference/completions/create
+        # available API params for Azure OpenAI: https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference#completions
         data = {
-            "model": model,
             "prompt": prompt,
             "temperature": temperature,
             "max_tokens": max_tokens,
             "top_p": top_p,
             "frequency_penalty": frequency_penalty,
-            "presence_penalty": presence_penalty
+            "presence_penalty": presence_penalty,
+            "best_of": best_of,
+            "stop": stop
+        }
+        if self.is_azure:  # Azure OpenAI
+            url_suffix = f"openai/deployments/{model}/completions?api-version={self.version}"
+        else:  # standard api.openai.com
+            data["model"] = model
+            url_suffix = "v1/completions"
+        return self._http_request(
+            method='POST', url_suffix=url_suffix, json_data=data,
+            headers=self.headers, resp_type='json', ok_codes=(200,)
+        )
+
+    def chatgpt(self, prompt: str, model: str = CHAT_DEFAULT_MODEL) -> dict:
+        """
+            Send prompt to ChatGPT using the 'chat/completions' endpoint.
+
+            Args:
+                prompt (str): Input to ChatGPT
+                model (str): The model which will generate the chat response
+
+            Returns:
+                dict: HTTP response from the OpenAI Chat API
+        """
+        options = {
+            "max_tokens": 1000,
```

Comment:
Is there a reason for specifying 1000 tokens exactly?

Reply:
Not really, again I was just leaving it from the existing `OpenAi ChatGPT v3 (Community Contribution)` integration. I think the logic is it is a good sanity limit to avoid costing customers too much money if they are just playing around with OpenAI.

Reply:
Does the API returns error when he reached the limit?

Reply:
Yes, if you try to pass in more than 1000 tokens it will throw a 400 error. I can make it a command argument like it is for `openai-completions`?

Reply:
Sound good!

Reply:
Done.